packages/synapse-react-client/src/synapse-client/SynapseClient.ts

```diff
@@ -1788,8 +1788,14 @@ export const signOut = async () => {
   await setAccessTokenCookie(undefined)
 }
 
+// Progress sent back during upload, will report when a part of a file is finished processing
+export type ProgressCallback = {
+  value: number
+  total: number
+}
+
 /**
- * Upload file.  Note that this currently only supports Synapse storage (Sage s3 bucket)
+ * Upload file to Synapse, creating a FileHandle
```

Comment:
Verified that I could use this code to upload to a custom storage location ID.
<img width="1380" alt="Screen Shot 2024-03-20 at 10 30 49 AM" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/1864447/09e32d9a-ef38-4d87-b51b-a81636c42e12">